x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/use_create_analytics_form.ts

```diff
@@ -125,25 +125,63 @@ export const useCreateAnalyticsForm = (): CreateAnalyticsFormProps => {
 
   const createKibanaIndexPattern = async () => {
     const dataViewName = destinationIndex;
+    let retryCount = 5;
 
     try {
-      await mlContext.dataViewsContract.createAndSave(
-        {
-          title: dataViewName,
-        },
-        false,
-        true
-      );
+      const interval = setInterval(async () => {
+        // Ran out of retry attempts
+        if (retryCount === 0) {
+          clearInterval(interval);
+          addRequestMessage({
+            error: 'No destination index check attempts left.',
+            message: i18n.translate(
+              'xpack.ml.dataframe.analytics.create.unableToCreateDataViewForDataFrameAnalyticsJob',
+              {
+                defaultMessage: 'Unable to create data view for:',
+              }
+            ),
+          });
+        }
 
-      addRequestMessage({
-        message: i18n.translate(
-          'xpack.ml.dataframe.analytics.create.createDataViewSuccessMessage',
-          {
-            defaultMessage: 'Kibana data view {dataViewName} created.',
-            values: { dataViewName },
+        try {
+          const resp = await ml.checkIndexExists(destinationIndex);
+          // Index exists - clear interval and create index pattern
+          if (resp && resp.exists) {
+            clearInterval(interval);
+            await mlContext.dataViewsContract.createAndSave(
+              {
+                title: dataViewName,
+              },
+              false,
+              true
+            );
+
+            addRequestMessage({
+              message: i18n.translate(
+                'xpack.ml.dataframe.analytics.create.createDataViewSuccessMessage',
+                {
+                  defaultMessage: 'Kibana data view {dataViewName} created.',
+                  values: { dataViewName },
+                }
+              ),
+            });
+          } else {
+            // Retry - index might not be created yet
+            retryCount -= 1;
           }
-        ),
-      });
+        } catch (e) {
+          clearInterval(interval);
+          addRequestMessage({
+            error: extractErrorMessage(e),
+            message: i18n.translate(
+              'xpack.ml.dataframe.analytics.create.errorCreatingDataViewForDataFrameAnalyticsJob',
+              {
+                defaultMessage: 'An error occurred creating the data view for:',
+              }
+            ),
+          });
+        }
+      }, 300);
```

Comment:
Trying 5 times with a delay of 300ms only gives ~1.5 seconds for the destination index to be created after kicking off the job start request, which isn't much. I ran into the timeout + error message a couple times during local tests.

Reply:
Good call - updated to max 15 tries with 1000ms between retries. e270e5e074d38a9a6c9861c45e2bebb836efbc9d